sky/clouds/aws.py

```diff
@@ -139,28 +139,28 @@ def get_default_ami(cls, region_name: str, instance_type: str) -> str:
     @classmethod
     def _get_image_id(
         cls,
-        image_id: Optional[Dict[str, str]],
+        image_id: Optional[Dict[Optional[str], str]],
         region_name: str,
-    ) -> str:
+    ) -> Optional[str]:
```

Comment:
Thinking more about this method, it's a little odd that `_get_image_id` can return None. If no image_id is specified, then this method should return a default image id (and thus the return type should be `str`).

In our current code, if image_id is None, we assign `image_id = self.get_default_ami(region_name, r.instance_type)` at L235. Instead, should we do that here at L146? Wdyt?